File: 1erdepartamental/python/11.matrices/main.py
"""
Programa: Matrices
Demuestra operaciones matriciales con operator overloading
"""

from core.matriz import Matriz

if __name__ == "__main__":
    m1 = Matriz(2, 2)
    m1.set(0, 0, 1)
    m1.set(0, 1, 2)
    m1.set(1, 0, 3)
    m1.set(1, 1, 4)

    print("Matriz original:")
    print(m1)

    print("\nMultiplicación escalar (x2):")
    print(m1 * 2)

    print("\nMultiplicación matricial:")
    print(m1 * m1)

File: 1erdepartamental/c++/11.matrices/main.cpp
#include <iostream>
#include "core/Matriz.h"

int main() {
    // Crea matriz 2x2
    Matriz m1(2, 2);
    m1.set(0, 0, 1); m1.set(0, 1, 2);
    m1.set(1, 0, 3); m1.set(1, 1, 4);
    
    std::cout << "Matriz original:" << std::endl << m1 << std::endl;
    
    // Multiplica por escalar
    Matriz m2 = m1 * 2;
    std::cout << "Multiplicación escalar (x2):" << std::endl << m2 << std::endl;
    
    // Multiplicacion matricial
    Matriz m3 = m1 * m1;
    std::cout << "Multiplicación matricial:" << std::endl << m3 << std::endl;
    
    return 0;
}
